numcodecs/ndarray_like.py

```diff
@@ -0,0 +1,56 @@
+import sys
+from typing import Any, Optional, Tuple
+
+if sys.version_info >= (3, 8):
+    from typing import Protocol, runtime_checkable
+else:
+    from typing_extensions import Protocol, runtime_checkable
```

Comment:
Think we will want to add `typing-extensions` to a few places

https://github.com/zarr-developers/numcodecs/blob/0b54e9c1cf2943dc9103c25955806c1174b04a65/setup.py#L339-L341

https://github.com/zarr-developers/numcodecs/blob/d16d1eac5198166a24726ffe808e3dcfcab9700d/requirements.txt#L2

https://github.com/zarr-developers/numcodecs/blob/a81856efbf8dfd8ea925acaac01235a096d285e4/requirements_dev.txt#L3

https://github.com/zarr-developers/numcodecs/blob/d16d1eac5198166a24726ffe808e3dcfcab9700d/requirements_rtfd.txt#L7

Reply:
Added